### PR TITLE
[ARTS-764] Authorise to retrieve only permission roles  the user is assigned to

### PR DESCRIPTION
**Pull request by LizaShak**
## Description
Restrict access to get permission roles. Authorise only if all the permission roles are roles that the user is assigned to.

### Changes
- [ARTS-764] - Get my roles
- Integrate security package in role service
- Add authorisation header on get roles by ids
- Authorise only if all the permission roles are roles that the user is assigned to

### Checklist:

- [ ] Branch name follows convention (feature/arts-#-short-name OR fix/arts-#-short-name)
- [ ] I have included a short-meaningful title, description and changes for this PR

[ARTS-764]: https://ndph-arts.atlassian.net/browse/ARTS-764

### CODE REVIEW

**Review by darrengsmithson — DISMISSED** [review]

**Comment by sameerapurini** [comment]
Hi @LizaShak I have added updates to api tests as part of my pr raised into yours, not sure why the same code has been copied over to your branch please

File: role-service-client/src/test/java/uk/ac/ox/ndph/mts/roleserviceclient/CreateEntityTest.java [hunk 1]
```diff
@@ -31,7 +31,7 @@
     public static MockWebServerWrapper webServer;
     private static final TestClientBuilder builder = new TestClientBuilder();
     private RoleServiceClient roleServiceClient;
-    private Consumer<HttpHeaders> authHeaders = RoleServiceClient.noAuth();
+    private String token = "some-token";
```
Comment (anatbal) [inline role-service-client/src/test/java/uk/ac/ox/ndph/mts/roleserviceclient/CreateEntityTest.java:34]:
```suggestion
    private String token = "some-token";
    private Consumer<HttpHeaders> authHeaders = RoleServiceClient.bearerAuth(token);
```
Then we can replace all RoleServiceClient.bearerAuth(token) with authHeaders

File: role-service-client/src/test/java/uk/ac/ox/ndph/mts/roleserviceclient/FindByIdTest.java [hunk 2]
```diff
@@ -26,7 +26,7 @@
     public static MockWebServerWrapper webServer;
     private static final TestClientBuilder builder = new TestClientBuilder();
     private RoleServiceClient roleServiceClient;
-    private Consumer<HttpHeaders> authHeaders = RoleServiceClient.noAuth();
+    private String token = "some-token";
```
Comment (anatbal) [inline role-service-client/src/test/java/uk/ac/ox/ndph/mts/roleserviceclient/FindByIdTest.java:29]: Same comment

File: role-service-client/src/test/java/uk/ac/ox/ndph/mts/roleserviceclient/FindByIdsTest.java [hunk 3]
```diff
@@ -27,7 +27,7 @@
     public static MockWebServerWrapper webServer;
     private static final TestClientBuilder builder = new TestClientBuilder();
     private RoleServiceClient roleServiceClient;
-    private Consumer<HttpHeaders> authHeaders = RoleServiceClient.noAuth();
+    private String token = "some-token";
```
Comment (anatbal) [inline role-service-client/src/test/java/uk/ac/ox/ndph/mts/roleserviceclient/FindByIdsTest.java:30]: Same comment

File: role-service-client/src/test/java/uk/ac/ox/ndph/mts/roleserviceclient/FindPageTest.java [hunk 4]
```diff
@@ -26,7 +26,7 @@
     public static MockWebServerWrapper webServer;
     private static final TestClientBuilder builder = new TestClientBuilder();
     private RoleServiceClient roleServiceClient;
-    private Consumer<HttpHeaders> authHeaders = RoleServiceClient.noAuth();
+    private String token = "some-token";
```
Comment (anatbal) [inline role-service-client/src/test/java/uk/ac/ox/ndph/mts/roleserviceclient/FindPageTest.java:29]: Same comment

File: role-service-client/src/test/java/uk/ac/ox/ndph/mts/roleserviceclient/UpdatePermissionsTest.java [hunk 5]
```diff
@@ -29,7 +29,7 @@
     public static MockWebServerWrapper webServer;
     private static final TestClientBuilder builder = new TestClientBuilder();
     private RoleServiceClient roleServiceClient;
-    private Consumer<HttpHeaders> authHeaders = RoleServiceClient.noAuth();
+    private final String token = "some-token";
```
Comment (anatbal) [inline role-service-client/src/test/java/uk/ac/ox/ndph/mts/roleserviceclient/UpdatePermissionsTest.java:32]: Same comment

**Review by anatbal — CHANGES_REQUESTED** [review]

**Review by anatbal — DISMISSED** [review]

**Review by anatbal — DISMISSED** [review]

**Review by darrengsmithson — DISMISSED** [review]

**Review by anatbal — APPROVED** [review]

**Review by darrengsmithson — APPROVED** [review]

**Pull request merged**